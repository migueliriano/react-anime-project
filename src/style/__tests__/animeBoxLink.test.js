import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import {
  LinkBoxAnime,
  BoxDetail,
  Title,
  Detail,
  Type,
  AverageRank,
} from '../animeBoxLink';


describe('BoxDetail style components', () => {
  it('Should render a LinkBoxAnime', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <LinkBoxAnime to="/" />
      </MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a BoxDetail Component', () => {
    const tree = renderer.create(<BoxDetail />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a Title Component', () => {
    const tree = renderer.create(<Title />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a Detail Component', () => {
    const tree = renderer.create(<Detail />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a Type Component', () => {
    const tree = renderer.create(<Type />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a AverageRank Component', () => {
    const tree = renderer.create(<AverageRank />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
