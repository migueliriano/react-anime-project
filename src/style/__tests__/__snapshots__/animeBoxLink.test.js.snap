// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxDetail style components Should render a AverageRank Component 1`] = `
.c0 {
  color: #53c4ff;
  font-weight: bold;
  text-transform: uppercase;
  margin-left: 15px;
}

<span
  className="c0"
/>
`;

exports[`BoxDetail style components Should render a BoxDetail Component 1`] = `
.c0 {
  background: #000;
  height: 5vw;
  width: 100%;
  position: absolute;
  bottom: 0;
  left: 0;
  background: linear-gradient(bottom,rgba(0,0,0,1),rgba(0,0,0,.4));
  background: -webkit-linear-gradient(bottom,rgba(0,0,0,1),rgba(0,0,0,.4));
  border-bottom-right-radius: 5px;
  border-bottom-left-radius: 5px;
  padding-top: 15px;
}

@media (max-width:768px) {
  .c0 {
    height: 10vw;
  }
}

@media (max-width:600px) {
  .c0 {
    height: 23vw;
  }
}

<div
  className="c0"
/>
`;

exports[`BoxDetail style components Should render a Detail Component 1`] = `
.c0 {
  position: absolute;
  bottom: 10px;
  left: 10px;
  font-size: 1.1vw;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2vw;
  }
}

@media (max-width:600px) {
  .c0 {
    font-size: 4vw;
  }
}

<div
  className="c0"
/>
`;

exports[`BoxDetail style components Should render a LinkBoxAnime 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  margin-right: 30px;
  margin-bottom: 30px;
  text-align: center;
  padding: 15px;
  width: 13vw;
  height: 20vw;
  background-size: cover;
  color: #fff;
  border-radius: 5px;
}

@media (max-width:768px) {
  .c0 {
    width: 33vw;
    height: 40vw;
  }
}

@media (max-width:600px) {
  .c0 {
    width: 100%;
    height: 90vw;
    margin-right: 0px;
  }
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
/>
`;

exports[`BoxDetail style components Should render a Title Component 1`] = `
.c0 {
  font-size: 1.1vw;
  font-weight: bold;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2vw;
  }
}

@media (max-width:600px) {
  .c0 {
    font-size: 5vw;
  }
}

<div
  className="c0"
/>
`;

exports[`BoxDetail style components Should render a Type Component 1`] = `
.c0 {
  color: #53c4ff;
  font-weight: bold;
  text-transform: uppercase;
}

<span
  className="c0"
/>
`;
